src/Microsoft.Dnx.Runtime/DependencyManagement/LibraryManager.cs

```diff
@@ -106,8 +130,14 @@ public void BuildInverseGraph()
 
         private void Visit(Library item,
                           Dictionary<string, List<Library>> inverse,
-                          HashSet<string> visited)
+                          HashSet<string> visited,
+                          Func<Library, bool> circularDependencyGuard)
         {
+            if (!circularDependencyGuard(item))
```

Comment:
Yeah, this was actually a problem in test code that was creating a circular dependency. I can remove the check now.